packages/perseus/src/widgets/free-response/free-response.tsx

```diff
@@ -45,17 +54,64 @@ export class FreeResponse
         };
     }
 
-    render(): React.ReactNode {
+    handleChange = (newValue: string) => {
+        this.setState({currentValue: newValue});
+    };
+
+    isOverLimit = () => {
+        return (
+            !this.props.allowUnlimitedCharacters &&
+            this.characterCount() > this.props.characterLimit
+        );
+    };
+
+    renderCharacterCount(): React.ReactNode {
```

Comment:
Just a style nit, but this feels a little inconsistent.

```TS
    isOverLimit = () => {
        // ...whatever
    }

    renderCharacterCount(): React.ReactNode {
        // ...whatever
    }
```

Is there a reason we couldn't stick to one format?

```suggestion
    isOverLimit() {
        return (
            !this.props.allowUnlimitedCharacters &&
            this.characterCount() > this.props.characterLimit
        );
    };

    renderCharacterCount(): React.ReactNode {
```